pkg/expression/builtin_time_test.go

```diff
@@ -2580,6 +2580,18 @@ func TestTimestampAdd(t *testing.T) {
 		{"MINUTE", 1.5, "1995-05-01 00:00:00", "1995-05-01 00:02:00"},
 		{"MINUTE", 1.5, "1995-05-01 00:00:00.000000", "1995-05-01 00:02:00"},
 		{"MICROSECOND", -100, "1995-05-01 00:00:00.0001", "1995-05-01 00:00:00"},
+
+		// issue41052
+		{"MONTH", 1, "2024-01-31", "2024-02-29 00:00:00"},
```

Comment:
We should've at least had one case for negative month.

Reply:
> We should've at least had one case for negative month.

fixed in https://github.com/pingcap/tidb/pull/54916